scim2-sdk-common/src/test/java/com/unboundid/scim2/common/BaseScimResourceTest.java

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright 2023-2025 Ping Identity Corporation
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License (GPLv2 only)
+ * or the terms of the GNU Lesser General Public License (LGPLv2.1 only)
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses>.
+ */
+
+package com.unboundid.scim2.common;
+
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.unboundid.scim2.common.exceptions.BadRequestException;
+import com.unboundid.scim2.common.messages.ErrorResponse;
+import com.unboundid.scim2.common.types.GroupResource;
+import com.unboundid.scim2.common.types.Meta;
+import com.unboundid.scim2.common.types.UserResource;
+import com.unboundid.scim2.common.utils.JsonUtils;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+
+/**
+ * Tests for {@link BaseScimResource}. Note that additional test coverage for
+ * schema extensions is included in the {@link ExtensionsTest}.
+ */
+public class BaseScimResourceTest
+{
+  /**
+   * Resets the customizable property toggled by this test class.
+   */
+  @AfterMethod
+  public void resetProperty()
+  {
+    BaseScimResource.IGNORE_UNKNOWN_FIELDS = false;
+  }
+
+  /**
+   * Test {@link BaseScimResource#setSchemaUrns}.
+   */
+  @Test
+  public void testBaseScimResourceSchemaUrns()
```

Comment:
These first two test methods were already added in the old SetSchemaTest. I've renamed this file to BaseScimResourceTest, which is why it appears again. I updated these to use AssertJ's methods over TestNG's assert methods for consistency.